mesonbuild/environment.py

```diff
@@ -105,13 +105,13 @@ def detect_llvm_cov(suffix: T.Optional[str] = None):
             tool = 'llvm-cov'
         else:
             tool = f'llvm-cov-{suffix}'
-        if mesonlib.exe_exists([tool, '--version']):
+        if shutil.which(tool) is not None:
```

Comment:
You can just do `if shutil.which(tool)` and in other cases "if not shutil.which(....)" -- the return value of shutil is always a positive string, or None if it wasn't found.